Community/Tdarr_Plugin_advo_FFMPEG_NVENC_Tiered_MKV_with_rollback.js

```diff
@@ -0,0 +1,563 @@
+      response.processFile = false;
+      response.infoLog += '☑File is already in hevc! \n';
+      return response;
+    }
+
+    if (file.video_resolution === '1080p' && (bitrateprobe / 1024) < inputs.hevcBitrateCutoff) {
+      response.processFile = false;
+      response.infoLog += '☑File is already in hevc! \n';
+      return response;
+    }
+    if (file.video_resolution === '4KUHD' && (bitrateprobe / 1024) < inputs.hevcBitrateCutoff * 1.5) {
+      response.processFile = false;
+      response.infoLog += '☑File is already in hevc! \n';
+      return response;
+    }
+
+    response.preset = '-c:v hevc_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'h264') {
+    if (bitrateprobe == null || bitrateprobe < 100000) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+    if ((file.video_resolution === '480p' || file.video_resolution === '576p')
+      && (bitrateprobe / 1024) < inputs.h264BitrateCutoff * 0.5) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+    if (file.video_resolution === '720p' && (bitrateprobe / 1024) < inputs.h264BitrateCutoff * 0.75) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+
+    if (file.video_resolution === '1080p' && (bitrateprobe / 1024) < inputs.h264BitrateCutoff) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+    if (file.video_resolution === '4KUHD' && (bitrateprobe / 1024) < inputs.h264BitrateCutoff * 1.5) {
+      response.processFile = false;
+      response.infoLog += '☑File is below cutoff already! \n';
+      return response;
+    }
+
+    if (parseInt(videoTrack.BitDepth, 10) > 8) {
+      // Remove HW Decoding for 10 bit
+    } else {
+      response.preset = '-c:v h264_cuvid';
+    }
+  } else if (file.video_codec_name.toLowerCase() === 'mjpeg') {
+    response.preset = 'c:v mjpeg_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'mpeg1') {
+    response.preset = '-c:v mpeg1_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'mpeg2') {
+    response.preset = '-c:v mpeg2_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'mpeg4') {
+    response.preset = '-c:v mpeg4_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vc1') {
+    response.preset = '-c:v vc1_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vp8') {
+    response.preset = '-c:v vp8_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vp9') {
+    response.preset = '-c:v vp9_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'av1') {
+    response.preset = '-c:v av1_cuvid';
+  } else if (file.video_codec_name.toLowerCase() === 'vc1') {
+    response.preset = '-c:v vc1_cuvid';
+  }
+
```

Comment:
Can be reduced down to `-hwaccel cuda`. This allows the GPU to decide if it's capable of HW decoding, if not it falls back to software.

Reply:
That does seem to work. I based this portion off of iiDrakeii's plugin. I assumed he had a good reason to do these checks.

Reply:
UPDATE: While it "works", it's significantly slower than the cuvid method I was using. My encodes went from about ~20 minutes to 5 hours when this was changed.